File: NapakalakiRuby/lib/player.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module NapakalakiGame

class Player
  
  MAXLEVEL = 10
  
  private
  def bringToLife
    @dead = false
  end
  
  private
  def getCombatLevel
    clevel = @level
    @visibleTreasures.each do |elemento|
      clevel += elemento.getBonus
    end
    return clevel
  end
  
  private
  def incrementLevels(l)
    @level += l
    if(@level > MAXLEVEL)
      @level = MAXLEVEL
    end
  end
  
  private
  def decrementLevels(l)
     @level -= l
    if(@level < 1)
      @level = 1
    end
  end
  
  private
  def setPendingBadConsequence(b)
    @pendingBadConsequence = b
  end
  
  private
  def applyPrize(m)
    #No se sabe
  end
  
  private
  def applyBadConsequence(m)
    #No se sabe
  end
  
  private
  def canMakeTreasureVisible(t)
    #No se sabe
  end
  
  private
  def howManyVisibleTreasures(tKind)
    cuantos=0;
    @visibleTreasures.each do |elemento|
      if elemento.getType == tKind
        cuantos+1
      end
    end
    return cuantos
  end
  
  private
  def dieIfNoTreasures
    if @visibleTreasures.empty? && @hiddenTreasures.empty?
      @dead = true
    end
  end
  
  private
  def giveMeATreasure
    #No se sabe
  end
  
  private
  def canYouGiveMeATreasure
    puedo=true;
    if @visibleTreasures.empty? && @hiddenTreasures.empty?
      puedo=false;
    end
    return puedo;
  end
  
  private
  def haveStolen
    @canISteal=false
  end

  def initialize(name)
    @name = name
    @level = 1
    @dead = true  
    @pendingBadConsequence = nil
    @visibleTreasures = Array.new
    @hiddenTreasures = Array.new
    @canISteal = true
    @enemy = nil
  end
  
  def getName
    return @name
  end
  
  def isDead
    return @dead
  end
  
  def getHiddenTreasures
    return @hiddenTreasures
  end
  
  def getVisibleTreasures
    return @visibleTreasures
  end
  
  def combat(m)
    #No se sabe
  end
  
  def makeTreasureVisible
    #No se sabe
  end
  
  def discardVisibleTreasure(t)
    #No se sabe
  end
  
  def discardHiddenTreasure(t)
    #No se sabe
  end
  
  def validState
    condicion = false;
    if @pendingBadConsequence.isEmpty && @hiddenTreasures.size < 5
      condicion = true;
    end
    return condicion;
  end
  
  def initTreasures
    #No se sabe
  end
  
  def getLevels
    return @level
  end
  
  def stealTreasure
    #No se sabe
  end
  
  def setEnemy(enemy)
    @enemy = enemy
  end
  
  def canISteal
    return @canISteal
  end
  
  def discardAllTreasures
    #No se sabe
  end
  
  def to_s
    cadena = "Nombre: #{@name}"
    cadena += "\nNivel: #{@level}"
    if @dead
      cadena += "\nEstá muerto"
    else
      cadena += "\nEstá vivo"
    end
    if @canISteal
      cadena += "\nPuede robar"
    else
      cadena += "\nNo puede robar"
      cadena += "\nEnemigo: #{@enemy}"
      cadena += "\nTesoros visibles: #{@visibleTreasures}"
      cadena += "\nTesoros ocultos: #{@hiddenTreasures}"
      cadena += "\nMal rollo a cumplir: #{@pendingBadConsequence}"
    end
    return cadena;    
  end
end

end

File: NapakalakiJava/src/NapakalakiGame/Player.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

import java.util.*;
/**
 *
 * @author danibolanos
 */
public class Player {
    static final int MAXLEVEL = 10;
    
    private String name;
    private int level;
    private boolean dead;
    private boolean canISteal;
    private Player enemy;
    private ArrayList<Treasure> visibleTreasures;
    private ArrayList<Treasure> hiddenTreasures;
    private BadConsequence pendingBadConsequence;
    
    private void bringToLife(){
        dead = false;
    }
    private int getCombatLevel(){
        int clevel = level;
        for(int i=0;i<visibleTreasures.size();i++)
            clevel+=visibleTreasures.get(i).getBonus();
        return clevel;
    }
    private void incrementLevels(int l){
        level+=l;
        if(level>MAXLEVEL)
            level=MAXLEVEL;
    }
    private void decrementLevels(int l){
        level-=l;
        if(level<1)
            level=1;
    }
    private void setPendingBadConsequence(BadConsequence b){
        pendingBadConsequence = b;
    }
    private void applyPrize(Monster m){
        //No se sabe
    }
    private void applyBadConsequence(Monster m){
        //No se sabe
    }
    private boolean canMakeTreasureVisible(Treasure t){
        //No se sabe
    }
    private int howManyVisibleTreasures(TreasureKind tKind){
        int cuantos=0;
        for(int i=0; i<visibleTreasures.size();++i)
            if(visibleTreasures.get(i).getType()==tKind)
                cuantos++;
        return cuantos;
    }
    private void dieIfNoTreasures(){
        if(visibleTreasures.isEmpty() && hiddenTreasures.isEmpty())
           dead = true;
    }
    private Treasure giveMeATreasure(){
        //No se sabe
    }
    private boolean canYouGiveMeATreasure(){
        boolean puedo=true;
        if(visibleTreasures.isEmpty() && hiddenTreasures.isEmpty())
            puedo=false;
        return puedo;
        
    }
    private void haveStolen(){
        canISteal=false;
    }
    public Player(String name){
        this.name=name;
        level=1;
        dead=true;
        canISteal = true;
        visibleTreasures = new ArrayList();
        hiddenTreasures = new ArrayList();
        pendingBadConsequence = null;
        enemy = null;
    }
    public String getName(){
        return name;
    }
    public boolean isDead(){
        return dead;
    }
    public ArrayList<Treasure> getHiddenTreasures(){
        //No se sabe
    }
    public ArrayList<Treasure> getVisibleTreasures(){
        //No se sabe
    }
    public CombatResult combat(Monster m){
        //No se sabe
    }
    public void makeTreasureVisible(Treasure t){
        //No se sabe
    }
    public void discardVisibleTreasure(Treasure t){
        //No se sabe
    }
    public void discardHiddenTreasure(Treasure t){
        //No se sabe
    }
    public boolean validState(){
        boolean condicion = false;
        if (pendingBadConsequence.isEmpty() && hiddenTreasures.size()<5)
            condicion = true;
        return condicion;
    }
    public void initTreasures(){
        //No se sabe
    }
    public int getLevels(){
        return level;
    }
    public Treasure stealTreasure(){
        //No se sabe
    }
    public void setEnemy(Player enemy){
        this.enemy=enemy;
    }
    public boolean canISteal(){
        return canISteal;
    }
    public void discardAllTreasures(){
        //No se sabe
    }
    public String toString(){
        String cadena = "Nombre: " + name;
        cadena += "\nNivel: " + level;
        if(dead)
           cadena += "\nEstá muerto";
        else
            cadena += "\nEstá vivo";
        if(canISteal)
            cadena += "\nPuede robar";
        else
            cadena += "\nNo puede robar";
        cadena += "\nEnemigo: " + enemy;
        cadena += "\nTesoros visibles: " + visibleTreasures.toString();
        cadena += "\nTesoros ocultos: " + hiddenTreasures.toString();
        cadena += "\nMal rollo a cumplir: " + pendingBadConsequence.toString();
        
        return cadena;
    }
}
